server/base/src/main/java/org/apache/accumulo/server/init/FileSystemInitializer.java

```diff
@@ -218,4 +193,38 @@ private void createEntriesForTablet(TreeMap<Key,Value> map, Tablet tablet) {
   private void addEntry(TreeMap<Key,Value> map, Text row, ColumnFQ col, Value value) {
     map.put(new Key(row, col.getColumnFamily(), col.getColumnQualifier(), 0), value);
   }
+
+  public String createScanRefTablet(ServerContext context, VolumeManager fs) throws IOException {
+    // First set table properties in ZK
+    setTableProperties(context, AccumuloTable.SCAN_REF.tableId(), initConfig.getScanRefTableConf());
+
+    VolumeChooserEnvironment chooserEnv =
+        new VolumeChooserEnvironmentImpl(VolumeChooserEnvironment.Scope.INIT,
+            AccumuloTable.METADATA.tableId(), SPLIT_POINT, context);
+
+    String tableMetadataTabletDirUri =
+        fs.choose(chooserEnv, context.getBaseUris()) + Constants.HDFS_TABLES_DIR + Path.SEPARATOR
+            + AccumuloTable.METADATA.tableId() + Path.SEPARATOR + TABLE_TABLETS_TABLET_DIR;
+
+    // Create Default Tablet entry for ScanServerRefTable
+    chooserEnv = new VolumeChooserEnvironmentImpl(VolumeChooserEnvironment.Scope.INIT,
+        AccumuloTable.SCAN_REF.tableId(), null, context);
+
+    String scanRefTableDefaultTabletDirName =
+        MetadataSchema.TabletsSection.ServerColumnFamily.DEFAULT_TABLET_DIR_NAME;
+    String scanRefTableDefaultTabletDirUri =
+        fs.choose(chooserEnv, context.getBaseUris()) + Constants.HDFS_TABLES_DIR + Path.SEPARATOR
+            + AccumuloTable.SCAN_REF.tableId() + Path.SEPARATOR + scanRefTableDefaultTabletDirName;
+
+    createDirectories(fs, tableMetadataTabletDirUri, scanRefTableDefaultTabletDirUri);
+
+    String ext = FileOperations.getNewFileExtension(DefaultConfiguration.getInstance());
+    String metadataFileName = tableMetadataTabletDirUri + Path.SEPARATOR + "0.1." + ext;
+
+    Tablet scanRefTablet =
+        new Tablet(AccumuloTable.SCAN_REF.tableId(), scanRefTableDefaultTabletDirName, null, null);
+    createMetadataFile(fs, metadataFileName, scanRefTablet);
+
+    return metadataFileName;
```

Comment:
The upgrade code is not doing anything with the return type of this method.  The init code will add a pointer to this file to another file the init code is creating.  Can not really do that sort of file manipulation  in the upgrade code, need to write to metadata table instead.  Howerver, I think these changes can be used with a few small tweaks starting with the following change to what is returned.

```suggestion
     InitialTablet scanRefTablet =
        new InitialTablet(AccumuloTable.SCAN_REF.tableId(), scanRefTableDefaultTabletDirName, null, null);
    return scanRefTablet;
```

With the above change in return type, hopefully could also do something like the following.  The goal being to be able to create key/values for an initial file in init code and a mutation for a metadata table update in upgrade code.

 * Rename `FileSystemInitializer.Tablet` to `FileSystemInitializer.InitialTablet` and make it public.  The rename is to make it more unique since it will be visible outside the class.
 * Create a method `TreeMap<Key,Value> createEntries()` in `InitialTablet` moving and refacotring the existing `FileSystemInitializer.createEntriesForTablet` into this new method.
 * Create a method `Mutation createMutation()` in  `InitialTablet` that calls `createEntries()` and converts to a mutation.
 * In the upgrade code use the returned `InitialTablet` to create a mutation and insert that into the metadata table.
 * In the init code use the returned `InitialTablet` to create key values entries in a file used for initial metadata. The init code is already doing this, would just be a refactoring.  Would be moving this file creation outside of the `createScanRefTablet()` method.